web/app/dao/SearchDAO.java

```diff
@@ -183,34 +164,94 @@
 	}
 
 	public static List<String> getAutoCompleteListDataset(String input, int limit) {
-		String pattern = isNotBlank(input) ? input + "%" : "%";
-		if (limit <= 0) {
-			return getJdbcTemplate().queryForList(GET_DATASET_AUTO_COMPLETE_LIST_NO_LIMIT, String.class, pattern);
-		}
-		return getJdbcTemplate().queryForList(GET_DATASET_AUTO_COMPLETE_LIST, String.class, pattern, limit);
+		String elasticSearchTypeURLKey = "elasticsearch.dataset.url";
+		String fieldName = "name_suggest";
+		return getAutoCompleteListbyES(elasticSearchTypeURLKey,fieldName,input,limit);
+	}
+	public static List<String> getAutoCompleteListMetric(String input, int limit) {
+		String elasticSearchTypeURLKey = "elasticsearch.metric.url";
+		String fieldName = "metric_name_suggest";
+		return getAutoCompleteListbyES(elasticSearchTypeURLKey,fieldName,input,limit);
 	}
-
 	public static List<String> getAutoCompleteListFlow(String input, int limit) {
-		String pattern = isNotBlank(input) ? input + "%" : "%";
-		List<String> names = new ArrayList<>();
-		if (limit <= 0) {
-			names.addAll(getJdbcTemplate().queryForList(GET_FLOW_AUTO_COMPLETE_LIST_NO_LIMIT, String.class, pattern));
-			names.addAll(getJdbcTemplate().queryForList(GET_JOB_AUTO_COMPLETE_LIST_NO_LIMIT, String.class, pattern));
-		} else {
-			names.addAll(getJdbcTemplate().queryForList(GET_FLOW_AUTO_COMPLETE_LIST, String.class, pattern, limit / 2));
-			names.addAll(getJdbcTemplate().queryForList(GET_JOB_AUTO_COMPLETE_LIST, String.class, pattern, limit / 2));
-		}
-		return names;
+		String elasticSearchTypeURLKey = "elasticsearch.flow.url";
+		String fieldName = "flow_name_suggest";
+		return getAutoCompleteListbyES(elasticSearchTypeURLKey,fieldName,input,limit);
 	}
 
-	public static List<String> getAutoCompleteListMetric(String input, int limit) {
-		String pattern = isNotBlank(input) ? input + "%" : "%";
-		if (limit <= 0) {
-			return getJdbcTemplate().queryForList(GET_METRIC_AUTO_COMPLETE_LIST_NO_LIMIT, String.class, pattern);
+	public static List<String> getAutoCompleteListbyES(String elasticSearchTypeURLKey, String fieldName, String input, int limit) {
+		// use elastic search completion suggester, ES will validate the input and limit
+		List<String> completionSuggestionList = new ArrayList<String>();
+		JsonNode responseNode = null;
+		ObjectNode keywordNode = null;
+
+		try
+		{
```

Comment:
Let's try keep the styling more consistent here. Use K&R (https://en.wikipedia.org/wiki/Indent_style#K.26R) like the rest of the code.